aspnetcore-webapi/RateLimitingDemo/RateLimitingDemo.UsingCustomMiddleware/Extensions/Serialization.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Newtonsoft.Json;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+
+namespace RateLimitingDemo.UsingCustomMiddleware.Extensions;
+
+public static class Serialization
+{
+    public static byte[] ToByteArray(this object objectToSerialize)
+    {
+        if (objectToSerialize == null)
+        {
+            return null;
+        }      
+
+        return Encoding.Default.GetBytes(JsonConvert.SerializeObject(objectToSerialize));
+    }
+
+    public static T FromByteArray<T>(this byte[] arrayToDeserialize) where T : class
+    {
+        if (arrayToDeserialize == null)
+        {
+            return default(T);
+        }      
+
+        return JsonConvert.DeserializeObject<T>(Encoding.Default.GetString(arrayToDeserialize));
+    }
```

Comment:
Just a confirmation, are these used anywhere?

Reply:
Yes. They are used in the DistributedCachingExtensions methods to get and set the cache value,